utils/check_python_imports.py

```diff
@@ -0,0 +1,182 @@
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+"""This script checks all Python imports within of provided files to align with
+our coding style, i.e., all internal imports are aliased with an underscore."""
+
+import ast as _ast
+import sys as _sys
+from dataclasses import dataclass as _dataclass
+from pathlib import Path as _Path
+from typing import Optional as _Optional
+
+
+@_dataclass
+class Error:
+    """Class to store import errors."""
+
+    filename: str
+    message: str
+    line_nr: _Optional[int] = None
+    column_nr: _Optional[int] = None
+
+
+class ImportChecker(_ast.NodeVisitor):
+    """Class to check if Python imports align with our coding style."""
+
+    def __init__(self, filename: str) -> None:
+        """Initialize the ImportChecker.
+
+        Args:
+            filename: The filename of the current file (to store for errors)
+        """
+
+        self.errors: list[Error] = []
+        self.filename = filename
+
+    def visit_Import(self, node: _ast.Import) -> None:
+        """Visit an import statement and check if the import aligns with our
+        coding style.
+
+        Args:
+            node: The import node to check.
+        """
+
+        for alias in node.names:
+            if alias.asname is None:
+                self.errors.append(
+                    Error(
+                        self.filename,
+                        f"Import '{alias.name}' has no alias!",
+                        node.lineno,
+                        node.col_offset,
+                    )
+                )
+            else:
+                if not alias.asname.startswith("_"):
+                    self.errors.append(
+                        Error(
+                            self.filename,
+                            f"Import '{alias.name}' has incorrect alias '{alias.asname}'; should start with '_'!",
+                            node.lineno,
+                            node.col_offset,
+                        )
+                    )
+        self.generic_visit(node)
+
+    def visit_ImportFrom(self, node: _ast.ImportFrom) -> None:
```

Comment:
```suggestion
    def visit_import_from(self, node: _ast.ImportFrom) -> None:
```

Or is this name required since we are overloading a method?

Reply:
This overloads the regular method - therefore, keeping the current name